internal/engine/container_build_and_deployer.go

```diff
@@ -60,15 +60,14 @@ func (cbd *ContainerBuildAndDeployer) BuildAndDeploy(ctx context.Context, servic
 
 	// ContainerBuildAndDeployer doesn't support initial build; call out to the ImageBuildAndDeployer
 	if state.IsEmpty() {
-		return cbd.ibd.BuildAndDeploy(ctx, service, state)
+		return BuildResult{}, fmt.Errorf("prev. build state is empty; container build does not support initial deploy")
 	}
 
 	// Otherwise, service has already been deployed; try to update in the running container
 
-	// (Unless we don't know what container it's running in, in which case fall back to image deploy.)
+	// (Unless we don't know what container it's running in, in which case we can't.)
 	if !state.LastResult.HasContainer() {
-		logger.Get(ctx).Infof("Unknown container, falling back to image deploy")
```

Comment:
did you intend to put this info message somewhere else?

Reply:
it now appears in the error message at right!